File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Map<Integer, Integer> newNums = new HashMap<>();
        for (int i = 0; i<nums.length; i++){
            newNums.put(nums[i], i);
        }
        for (int i = 0; i<nums.length; i++){
            if(newNums.get(nums[i]) != i){
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        final = len(nums) - 1
        while start <= final:
            m = start + ((final - start) // 2)
            if target > nums[m]:
                start = m+1
            elif target < nums[m]:
                final = m -1 
            else:
                 return m

        return -1;

